react/week2/todo-part2/src/App.js

```diff
@@ -0,0 +1,128 @@
+import React, { useState, useEffect } from "react";
+import ToDos from "./ToDos";
+
+const Header = () => {
+  return <h1> Todo List</h1>;
+};
+
+const Timer = () => {
+  const [count, setCount] = useState(0);
+  useEffect(() => {
+    setTimeout(() => {
+      setCount((timer) => timer + 1);
+    }, 1000);
+  });
```

Comment:
Great idea and so close but use the setInterval (which runs at every interval of e.g. 1000)

setTimeout runs once delayed by 1000

It may appear to work but that is because your component keeps re-rendering ever triggering the delay again and again.

That's why it is import you cleanup this onmount useEffect
```
useEffect(() => {
    const timer = setInterval(() => {
      setCount((timer) => timer + 1);
    }, 1000);
    return () => clearInterval(timer)
  }, [);  // <---- also remember the onmount []
```

Reply:
thanks for all the comments I will change them